deploy/Dockerfile

```diff
@@ -17,31 +17,56 @@
 FROM golang:1.14
 ARG GOARCH=amd64
 WORKDIR /go/src/github.com/GoogleContainerTools/kaniko
+
+RUN echo $GOARCH > /goarch
```

Comment:
no need to write this to file if you to using `TARGETARCH`

Reply:
I did this as multiple image layers are created and ARG is not preserved across layers.